recipes-security/optee-imx/optee-os_3.2.0.imx.bb

```diff
@@ -71,8 +71,8 @@ do_deploy () {
 }
 
 do_install () {
-    install -d ${D}/lib/firmware/
-    install -m 644 ${B}/core/*.bin ${D}/lib/firmware/
+    install -d ${D}/usr/lib/firmware/
+    install -m 644 ${B}/core/*.bin ${D}/usr/lib/firmware/
```

Comment:
The fix in fact should be on the packaging. It needs to include the `/lib/firmware` on `FILES_${PN}`.

You should use is, likely, `${base_libdir}/firmware`.

Another issue with the commit, is the commit log. It should follow https://www.openembedded.org/wiki/How_to_submit_a_patch_to_OpenEmbedded#Committing_your_patch

@thochstein please take a look here as we likely need to fix (or backport a fix) on master.

@antznin check if dunfell branch works for you, please.

Reply:
`${base_libdir}` is actually `/usr/lib`, so that won't help fix the current `FILES_${PN}` which is currently `${nonarch_base_libdir}/firmware`.

The problem does not persist when using `FILES_${PN} = "${base_libdir_native}/firmware"`. If that's okay, I will reset my commit, do the changes, and properly sign it. Thanks for pointing at that btw.

Reply:
Actually, sorry, but this change prompts me an error, which I didn't notice : 
```
ERROR: optee-os-3.2.0.imx-r0 do_package_qa: QA Issue: optee-os package is not obeying usrmerge distro feature. /lib should be relocated to /usr. [usrmerge]
```

I might have misunderstood your answer, since I am not yet super familiar with Yocto. Please, would you mind reformulate your fix?

Reply:
> `${base_libdir}` is actually `/usr/lib`, so that won't help fix the current `FILES_${PN}` which is currently `${nonarch_base_libdir}/firmware`.

By default, `base_libdir` points to `/lib`. It points to `/usr/lib` only when you use `usrmerge` DISTRO_FEATURE`.

> The problem does not persist when using `FILES_${PN} = "${base_libdir_native}/firmware"`. If that's okay, I will reset my commit, do the changes, and properly sign it. Thanks for pointing at that btw.

The fix should be done on the `do_install` task. Try the diff below:

```
diff --git a/recipes-security/optee-imx/optee-os_3.2.0.imx.bb b/recipes-security/optee-imx/optee-os_3.2.0.imx.bb
index ba903519..06563a18 100644
--- a/recipes-security/optee-imx/optee-os_3.2.0.imx.bb
+++ b/recipes-security/optee-imx/optee-os_3.2.0.imx.bb
@@ -73,8 +73,8 @@ do_deploy () {
 }

 do_install () {
-    install -d ${D}/lib/firmware/
-    install -m 644 ${B}/core/*.bin ${D}/lib/firmware/
+    install -d ${D}${nonarch_base_libdir}/firmware/
+    install -m 644 ${B}/core/*.bin ${D}${nonarch_base_libdir}/firmware/

     # Install the TA devkit
     install -d ${D}/usr/include/optee/export-user_ta_${OPTEE_ARCH}/
```

Reply:
We need to check if master also need this fix. Could you check?

Reply:
Yes, this also happens on master as of today. The same fix solves the issue.